library/cwm/test/common_widgets_test.rb

```diff
@@ -59,3 +58,47 @@ def hspacing
     end
   end
 end
+
+describe CWM::RichText do
+  subject { described_class.new }
+  let(:widget_id) { Id(subject.widget_id) }
+
+  describe "#value=" do
+    before do
+      allow(subject).to receive(:keep_scroll?).and_return(keep_scroll)
+      allow(Yast::UI).to receive(:ChangeWidget)
+    end
+
+    context "when set to restore the scroll" do
+      let(:keep_scroll) { true }
+
+      it "saves the scroll position" do
+        expect(Yast::UI).to receive(:QueryWidget).with(widget_id, :VScrollValue)
+
+        subject.value = "Test"
+      end
+
+      it "restores the scroll" do
+        expect(Yast::UI).to receive(:ChangeWidget).with(widget_id, :VScrollValue, anything)
+
+        subject.value = "Test"
+      end
+    end
+
+    context "when set to not restore the scroll" do
+      let(:keep_scroll) { false }
+
+      it "saves the scroll position" do
```

Comment:
is this important behavior? Why testing it as it is lost? I think we should not test this behavior as we can easily remove it in future.

Reply:
> is this important behavior? Why testing it as it is lost?

I don't understand these questions. What's the problem adding unit testing here?

>  I think we should not test this behavior as we can easily remove it in future.

I get confused. If we remove this option in the future (!?), we simply can remove the associated tests. Can't we?

Reply:
> > is this important behavior? Why testing it as it is lost?
> 
> I don't understand these questions. What's the problem adding unit testing here?

Well, rspec idea is to tests specification. And saving scroll option is not visible from outside and it is also not interested for user (of API) as he cannot use it. So it makes more sense for me to have specification like `it "does not modify vertical scrolling" do`

> 
> > I think we should not test this behavior as we can easily remove it in future.
> 
> I get confused. If we remove this option in the future (!?), we simply can remove the associated tests. Can't we?

question is does behavior of method change if we no longer store scrolling? My issue here is that test is very tight with exact implementation and so very fragile. Ideally in rspec if you change implementation you should change test content, but not specification written in tests. That is difference between unit testing and behavior testing.

Reply:
> I don't understand these questions. What's the problem adding unit testing here?

Ok, I think I got it now, after seeing the comment in the full context. You mean testing that the vscroll has been saved when it will be not used and it is not public.

I think it does not hurt, but I'll remove it and try to improve the other examples.

Reply:
Ups, I hadn't seen your comment until refresh the page.

Thanks!